ring/ring.go

```diff
@@ -237,31 +237,31 @@ func NewWithStoreClientAndStrategy(cfg Config, name, key string, store kv.Client
 		ringDesc:             &Desc{},
 		shuffledSubringCache: map[subringCacheKey]*Ring{},
 		memberOwnershipDesc: prometheus.NewDesc(
-			"cortex_ring_member_ownership_percent",
+			"ring_member_ownership_percent",
```

Comment:
These metrics are exposed by `ring.Collect()`. How are we going to add the prefix back in Cortex? What's the plan?

Reply:
Seems like this could be handled with something like `prometheus.WrapRegistererWithPrefix("cortex_", reg)` before calling `reg.MustRegister(ring)`. Am I missing something or are you specifically asking if there's an open issue / design for doing that?

Reply:
Chatted offline with @treid314, I now understand why this is not as easy as it seems.

Reply:
There is no promauto.With(registerer).NewDisc, should we move to a Summary here? We already have the const labels for these metrics on register time.

Reply:
I think an option is to remove the `Describe()` and `Collect()` functions from the `Ring` and move all these metrics to `promauto.With(reg).NewGaugeFunc()`, setting the name label as a const label (whose value is is the ring `name`). Compared to `NewGauge()`, the `NewGaugeFunc()` takes a function which is called each time Prometheus client needs to collect the metric value. The current logic inside `Collect()` can be moved to separate functions like `getMemberOwnershipMetric() float64`, `getNumMembersMetric() float64`, and so on.

Reply:
_Tyler if the suggestion is not clear, please reach out and I can provide some more guidance._

Reply:
Talked with @pracucci offline, there's not really a good way to use GaugeFuncs here since we have a constant and variable label for these metrics. I'm going to look into using a ticker to update GaugeVecs at a specified interval for the ring metrics.